src/test/clip.t.sol

```diff
@@ -302,6 +302,27 @@ contract ClipperTest is DSTest {
         assertEq(vat.dai(bob), rad(1000 ether) + rad(100 ether) + due * 0.02 ether / WAD); // Paid (tip + due * chip) amount of DAI for calling bark()
     }
 
+    function try_kick(uint256 tab, uint256 lot, address usr) internal returns (bool ok) {
+        string memory sig = "kick(uint256,uint256,address)";
+        (ok,) = address(clip).call(abi.encodeWithSignature(sig, tab, lot, usr));
+    }
+
+    function test_kick_basic() public {
+        assertTrue(try_kick(1 ether, 2 ether, address(1)));
+    }
+
+    function test_kick_zero_tab() public {
+        assertTrue(!try_kick(0, 2 ether, address(1)));
+    }
+
+    function test_kick_zero_lot() public {
+        assertTrue(!try_kick(1 ether, 0, address(1)));
+    }
+
+    function test_kick_zero_usr() public {
+        assertTrue(!try_kick(1 ether, 2 ether, address(0)));
+    }
```

Comment:
We should add this test for `take()` and `redo()` as well

Reply:
Done. Also converted a couple `testFail` tests for `redo` to be just `test` with different assertions (since that is less fragile).